manager/controlapi/service.go

```diff
@@ -327,7 +327,9 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 			return nil
 		}
 		// temporary disable network update
-		if request.Spec != nil && !reflect.DeepEqual(request.Spec.Networks, service.Spec.Networks) {
+		if request.Spec != nil &&
+			(!reflect.DeepEqual(request.Spec.Task.Networks, service.Spec.Task.Networks) ||
+				!reflect.DeepEqual(request.Spec.Networks, service.Spec.Networks)) {
```

Comment:
Hmm... Ok. If you want to provide users a way to migrate their services created in old model to new model. Let me rework the code to accommodate that case.